frontend/src/app/spot/components/text-field/stories/TextField.stories.mdx

```diff
@@ -46,33 +52,83 @@ export const Template = (args) => ({
   },
 })
 
+
+The text field component affords a few optional elements:
+
+An optional icon can be added to the left edge to provide additional context when this is necessary (in a search field, for example):
+
+> Example:  A search bar with a icon to the left
+
+An icon can also optionally be placed on the right edge. This arrow can provide additional hints on how the text input field will behave; for example, a right arrow might suggest that the value will be immediately submitted, or a down arrow might indicate that the text field will open a drop modal underneath it (a list of assignees, for example).
+
+The right icon can also be converted to an icon-only link that allows the user to clear the field.
+
+> Example: A text field with a search icon, preset text and a clear icon on the right
+
+## Behaviour
+
+Clicking on a text field will move it to the focused state. 
+
+A text field can have placeholder text to provide additionanl context or information; this text is replaced by actual text that the user inputs, and is shown again only if the field is cleared.
+
 <Canvas>
   <Story
-    name="Preset value"
+    name="Placeholder"
     args={{
-      value: 'I have a value set',
+      placeholder: 'Placeholders get cut off if they\'re too long',
     }}
   >
     {Template.bind({})}
   </Story>
 </Canvas>
+
+The clear icon-only link on the right only appears on some text fields.
+
+The width of a text field is fixed and defined either manually, or by the width of the containing element. When text exceeds the available width, it scrolls with the cursor within the text field (and clips on the other end).
+
+## States
+
+The **Default** state indicates that a text field is ready to accept text input.
+
+
 <Canvas>
   <Story
-    name="Disabled"
+    name="Preset value"
     args={{
-      disabled: true,
+      value: 'I have a value set',
     }}
   >
     {Template.bind({})}
   </Story>
 </Canvas>
+
+The **Focus** state has a coloured outline, the same as any focused element in OpenProject.
+
+The **Hover** element has a darker border.
+
+The **Disabled** state shows all elements (including icons and any placeholder or input text) in grey, and makes any interaction impossible.
```

Comment:
Should the grey and darker border be specified? Like which colour hex are them?

Reply:
Yep, I added it. But I'm never sure I should mention them explicitly, since we haven't _actually_ implemented our named colours yet in Storybook.